kubernetes/storage.yaml

```diff
@@ -73,6 +77,14 @@ spec:
       resources:
         requests:
           storage: 1Gi
+  - metadata:
```

Comment:
😅 pretty new to k8's manifest (using microk8s)

Running into some errors with the deployment and was wondering what the previous workflow was for validating these yaml files? `kubeval` seems to indicate no issues. 

```
$ kubeval storage.yaml
PASS - storage.yaml contains a valid StatefulSet (buildbarn.storage)
PASS - storage.yaml contains a valid Service (buildbarn.storage)
```
This error seems to occur after adding this additional metadata section.
Deployment fails. I was taking a look at [docs](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) however, I couldn't seem to find anything that seemed out of the ordinary for declaring persistent volume mounts?


Would appreciate a second pair of eyes 😅 

To repro.
when running `kubectl apply -k .`, 
```
Error from server (Invalid): error when applying patch:
{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"apps/v1\",\"kind\":\"Deployment\",\"metadata\":{\"annotations\":{\"prometheus.io/port\":\"80\",\"prometheus.io/scrape\":\"true\"},\"name\":\"worker-ubuntu22-04\",\"namespace\":\"buildbarn\"},\"spec\":{\"replicas\":8,\"selector\":{\"matchLabels\":{\"app\":\"worker\",\"instance\":\"ubuntu22-04\"}},\"template\":{\"metadata\":{\"labels\":{\"app\":\"worker\",\"instance\":\"ubuntu22-04\"}},\"spec\":{\"containers\":[{\"args\":[\"/config/worker-ubuntu22-04.jsonnet\"],\"env\":[{\"name\":\"NODE_NAME\",\"valueFrom\":{\"fieldRef\":{\"fieldPath\":\"spec.nodeName\"}}},{\"name\":\"POD_NAME\",\"valueFrom\":{\"fieldRef\":{\"fieldPath\":\"metadata.name\"}}}],\"image\":\"buildbarn/bb-worker:20221223T045242Z-9de9f27\",\"name\":\"worker\",\"volumeMounts\":[{\"mountPath\":\"/config/\",\"name\":\"configs\",\"readOnly\":true},{\"mountPath\":\"/worker\",\"name\":\"worker\"},{\"mountPath\":\"/storage-worker-cas\",\"name\":\"storage-worker-cas\"}]},{\"command\":[\"/bb/tini\",\"-v\",\"-g\",\"--\",\"/bb/bb_runner\",\"/config/runner-ubuntu22-04.jsonnet\"],\"image\":\"ghcr.io/catthehacker/ubuntu:act-22.04@sha256:5f9c35c25db1d51a8ddaae5c0ba8d3c163c5e9a4a6cc97acd409ac7eae239448\",\"name\":\"runner\",\"securityContext\":{\"allowPrivilegeEscalation\":false,\"runAsUser\":65534},\"volumeMounts\":[{\"mountPath\":\"/config/\",\"name\":\"configs\",\"readOnly\":true},{\"mountPath\":\"/worker\",\"name\":\"worker\"},{\"mountPath\":\"/storage-worker-cas\",\"name\":\"storage-worker-cas\"},{\"mountPath\":\"/bb\",\"name\":\"bb-runner\",\"readOnly\":true}]}],\"initContainers\":[{\"image\":\"buildbarn/bb-runner-installer:20221223T045242Z-9de9f27\",\"name\":\"bb-runner-installer\",\"volumeMounts\":[{\"mountPath\":\"/bb/\",\"name\":\"bb-runner\"}]},{\"command\":[\"sh\",\"-c\",\"mkdir -pm 0777 /worker/build \\u0026\\u0026 mkdir -pm 0700 /worker/cache \\u0026\\u0026 mkdir -pm 0777 /storage-worker-cas \\u0026\\u0026 chmod 0777 /worker\"],\"image\":\"busybox:1.31.1-uclibc\",\"name\":\"volume-init\",\"volumeMounts\":[{\"mountPath\":\"/worker\",\"name\":\"worker\"},{\"mountPath\":\"/storage-worker-cas\",\"name\":\"storage-worker-cas\"}]}],\"volumes\":[{\"emptyDir\":{},\"name\":\"bb-runner\"},{\"name\":\"configs\",\"projected\":{\"sources\":[{\"configMap\":{\"items\":[{\"key\":\"runner-ubuntu22-04.jsonnet\",\"path\":\"runner-ubuntu22-04.jsonnet\"}],\"name\":\"runner-ubuntu22-04\"}},{\"configMap\":{\"items\":[{\"key\":\"worker-ubuntu22-04.jsonnet\",\"path\":\"worker-ubuntu22-04.jsonnet\"}],\"name\":\"worker-ubuntu22-04\"}},{\"configMap\":{\"items\":[{\"key\":\"common.libsonnet\",\"path\":\"common.libsonnet\"}],\"name\":\"common\"}}]}},{\"emptyDir\":{},\"name\":\"worker\"}]}}}}\n"}},"spec":{"template":{"spec":{"$setElementOrder/containers":[{"name":"worker"},{"name":"runner"}],"$setElementOrder/initContainers":[{"name":"bb-runner-installer"},{"name":"volume-init"}],"containers":[{"$setElementOrder/volumeMounts":[{"mountPath":"/config/"},{"mountPath":"/worker"},{"mountPath":"/storage-worker-cas"}],"image":"buildbarn/bb-worker:20221223T045242Z-9de9f27","name":"worker","volumeMounts":[{"mountPath":"/storage-worker-cas","name":"storage-worker-cas"}]},{"$setElementOrder/volumeMounts":[{"mountPath":"/config/"},{"mountPath":"/worker"},{"mountPath":"/storage-worker-cas"},{"mountPath":"/bb"}],"image":"ghcr.io/catthehacker/ubuntu:act-22.04@sha256:5f9c35c25db1d51a8ddaae5c0ba8d3c163c5e9a4a6cc97acd409ac7eae239448","name":"runner","volumeMounts":[{"mountPath":"/storage-worker-cas","name":"storage-worker-cas"}]}],"initContainers":[{"$setElementOrder/volumeMounts":[{"mountPath":"/worker"},{"mountPath":"/storage-worker-cas"}],"command":["sh","-c","mkdir -pm 0777 /worker/build \u0026\u0026 mkdir -pm 0700 /worker/cache \u0026\u0026 mkdir -pm 0777 /storage-worker-cas \u0026\u0026 chmod 0777 /worker"],"name":"volume-init","volumeMounts":[{"mountPath":"/storage-worker-cas","name":"storage-worker-cas"}]}]}}}}
to:
Resource: "apps/v1, Resource=deployments", GroupVersionKind: "apps/v1, Kind=Deployment"
Name: "worker-ubuntu22-04", Namespace: "buildbarn"
for: ".": error when patching ".": Deployment.apps "worker-ubuntu22-04" is invalid: [spec.template.spec.containers[0].volumeMounts[2].name: Not found: "storage-worker-cas", spec.template.spec.containers[1].volumeMounts[2].name: Not found: "storage-worker-cas", spec.template.spec.initContainers[1].volumeMounts[1].name: Not found: "storage-worker-cas"]
Error from server (Invalid): error when applying patch:
{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"apps/v1\",\"kind\":\"StatefulSet\",\"metadata\":{\"annotations\":{},\"name\":\"storage\",\"namespace\":\"buildbarn\"},\"spec\":{\"replicas\":2,\"selector\":{\"matchLabels\":{\"app\":\"storage\"}},\"serviceName\":\"storage\",\"template\":{\"metadata\":{\"labels\":{\"app\":\"storage\"}},\"spec\":{\"containers\":[{\"args\":[\"/config/storage.jsonnet\"],\"image\":\"buildbarn/bb-storage:20221221T163740Z-08ce6e4\",\"name\":\"storage\",\"ports\":[{\"containerPort\":8981,\"protocol\":\"TCP\"}],\"volumeMounts\":[{\"mountPath\":\"/config/\",\"name\":\"configs\",\"readOnly\":true},{\"mountPath\":\"/storage-cas\",\"name\":\"cas\"},{\"mountPath\":\"/storage-ac\",\"name\":\"ac\"},{\"mountPath\":\"/storage-fsac\",\"name\":\"fsac\"}]}],\"initContainers\":[{\"command\":[\"sh\",\"-c\",\"mkdir -m 0700 -p /storage-cas/persistent_state /storage-ac/persistent_state /storage-fsac/persistent_state\"],\"image\":\"busybox:1.31.1-uclibc\",\"name\":\"volume-init\",\"volumeMounts\":[{\"mountPath\":\"/storage-cas\",\"name\":\"cas\"},{\"mountPath\":\"/storage-ac\",\"name\":\"ac\"},{\"mountPath\":\"/storage-fsac\",\"name\":\"fsac\"}]}],\"volumes\":[{\"name\":\"configs\",\"projected\":{\"sources\":[{\"configMap\":{\"items\":[{\"key\":\"storage.jsonnet\",\"path\":\"storage.jsonnet\"}],\"name\":\"storage\"}},{\"configMap\":{\"items\":[{\"key\":\"common.libsonnet\",\"path\":\"common.libsonnet\"}],\"name\":\"common\"}}]}}]}},\"volumeClaimTemplates\":[{\"metadata\":{\"name\":\"cas\"},\"spec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"12Gi\"}}}},{\"metadata\":{\"name\":\"ac\"},\"spec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"1Gi\"}}}},{\"metadata\":{\"name\":\"fsac\"},\"spec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"12Gi\"}}}}]}}\n"}},"spec":{"replicas":2,"template":{"spec":{"$setElementOrder/containers":[{"name":"storage"}],"containers":[{"image":"buildbarn/bb-storage:20221221T163740Z-08ce6e4","name":"storage"}]}},"volumeClaimTemplates":[{"metadata":{"name":"cas"},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"12Gi"}}}},{"metadata":{"name":"ac"},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"1Gi"}}}},{"metadata":{"name":"fsac"},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"12Gi"}}}}]}}
to:
Resource: "apps/v1, Resource=statefulsets", GroupVersionKind: "apps/v1, Kind=StatefulSet"
Name: "storage", Namespace: "buildbarn"
for: ".": error when patching ".": StatefulSet.apps "storage" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden
```

Reply:
Resolved via recreating and deleting 🤦